Source/AwsCommonRuntimeKit/auth/credentials/AWSCredentialsProvider.swift

```diff
@@ -330,22 +330,23 @@ extension AWSCredentialsProvider.Source {
             x509Options.bootstrap = bootstrap.rawValue
             x509Options.shutdown_options = shutdownCallbackCore.getRetainedCredentialProviderShutdownOptions()
 
-            guard let provider: UnsafeMutablePointer<aws_credentials_provider> = (withByteCursorFromStrings(
-                thingName,
-                roleAlias,
-                endpoint) { thingNameCursor, roleAliasCursor, endPointCursor in
-
-                x509Options.thing_name = thingNameCursor
-                x509Options.role_alias = roleAliasCursor
-                x509Options.endpoint = endPointCursor
-                return withOptionalCStructPointer(
-                    proxyOptions,
-                    tlsConnectionOptions) { proxyOptionsPointer, tlsConnectionOptionsPointer in
-
-                    x509Options.proxy_options = proxyOptionsPointer
-                    x509Options.tls_connection_options = tlsConnectionOptionsPointer
-                    return aws_credentials_provider_new_x509(allocator.rawValue, &x509Options)
-                }})
+            guard let provider: UnsafeMutablePointer<aws_credentials_provider> = (
```

Comment:
Does `withByteCursorFromStrings` need to be wrapped in parentheses?

Reply:
Yes, we need to need wrap withByteCursorFromStrings or not use trailing closure.

Reply:
In that case, its preferable to not use the trailing closure

Reply:
Thanks, fixed.